--- a/src/backend/tty.rs
+++ b/src/backend/tty.rs
@@ -522,7 +522,7 @@
             .add_node(render_node, gbm.clone())
             .context("error adding render node to GPU manager")?;
 
-        if node == self.primary_node || render_node == self.primary_render_node {
+        if (node == self.primary_node || render_node == self.primary_render_node) && self.dmabuf_global.is_none() {
             if node == self.primary_node {
                 debug!("this is the primary node");
             }
